### ISSUE TITLE
x/tools/gopls: don't warn about misconfiguration in std and cmd modules

### ISSUE THREAD

**Issue by FiloSottile** [golang/go#37906]
I am using the latest gopls while developing the standard library (specifically `crypto/x509`) at tip, and everything seems to work, but the following warning keeps popping up, presumably because the `std` module is weird, but still works with `go/packages`.

> You are neither in a module nor in your GOPATH. Please see https://github.com/golang/go/wiki/Modules for information on how to set up your Go project.

```
$ go version
go version devel +971f8a2f9a Tue Mar 17 23:02:30 2020 +0000 darwin/amd64
```

**Comment by stamblerre** [comment]
What directory are you opening in VS Code?

**Comment by FiloSottile** [comment]
I am opening the GOROOT. Should I be opening `GOROOT/std`?

That does make sense, but it's pretty inconvenient for standard library development, because the outer directories have non-go files that are useful to look at the same time.

**Comment by zikaeroh** [comment]
I recently submitted a few CLs and used `gopls` for it; to make things (kinda) work, I had to both open `GOROOT/src`, and ensure that `GOROOT/bin` was in my `PATH` when I launched my editor (otherwise things mismatched).

**Comment by bcmills** [comment]
`GOROOT` is not in the `std` module, but it should be ok to build and test things in `std` and `cmd` without being in those modules (as long as you don't need to update their vendored dependencies).

**Comment by stamblerre** [comment]
This is a current limitation of `gopls` - we're aiming to fix this in [`gopls/v0.5.0`](https://github.com/golang/go/milestone/138). You should only see this error message if you open `$GOROOT` and add an unresolved import, but in the meantime, the recommended work-around is to add the folder to your workspace (File -> Add folder to workspace). I'll close this issue now, but we'll hopefully have it resolved by `gopls/v0.5.0`.
